### DISCUSSION TITLE
How to extend components default properties to be used in theme

### DISCUSSION THREAD

**Discussion by cflorez10** [react-native-elements/react-native-elements#3371] [Q&A]
What I want is to extend default properties for buttons, so what I do is extend the default `ButtonProps` like so:

```typescript
import 'patch/to/node_modules/react-native-elements';

declare module 'react-native-elements' {
  export interface ButtonProps {
    size: string;
  }

  export interface FullTheme {
    Button: Partial<ButtonProps>;
  }
}
```

With this I want to pass the new properties in `theme` file like this: 

```typescript
import { ThemeOptions, createTheme } from 'react-native-elements';

export const theme: ThemeOptions = createTheme({ 
  Button: {
    titleStyle: {
      fontFamily: Font.primary.bold,
    },
    size: ''
  },
})
```

this is not working, I'm getting the following error:

```shell
TS2322: Type '{ titleStyle: { fontFamily: string; }; size: string; }' is not assignable to type 'Partial<ButtonProps>'.   Object literal may only specify known properties, and 'size' does not exist in type 'Partial<ButtonProps>'.
```

- is it possible to increase the properties that a component receives in this way or am I doing something wrong?
- If this is not posible what is the recomended way to handle properties like size(small, medium, large) or type(gost, outlined, filled) for components like buttons

**Comment by arpitBhalla** [comment]
You don't need to do use `ThemeOptions`, `createTheme` take care of types
```diff
- const theme: ThemeOptions = createTheme({ 
+ const theme = createTheme({ 
```
